core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -325,12 +325,18 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
 
   override def getListing(): Iterator[ApplicationInfo] = {
     // Return the listing in end time descending order.
-    listing.view(classOf[ApplicationInfoWrapper])
+    // SPARK-38896: tryWithResource cannot be used here
+    // because this method needs to return an `Iterator`.
+    val closeableIter = listing.view(classOf[ApplicationInfoWrapper])
```

Comment:
`tryWithResource` cannot be used here because this method needs to return an `Iterator`,  mixed in `java.io.Closeable` so that the caller can close it